src/ophyd_async/epics/areadetector/drivers/ad_base.py

```diff
@@ -18,11 +37,29 @@ def __init__(self, prefix: str, name: str = "") -> None:
         self.array_counter = ad_rw(int, prefix + "ArrayCounter")
         self.array_size_x = ad_r(int, prefix + "ArraySizeX")
         self.array_size_y = ad_r(int, prefix + "ArraySizeY")
+        self.detector_state = ad_r(DetectorState, prefix + "DetectorState")
         # There is no _RBV for this one
         self.wait_for_plugins = epics_signal_rw(bool, prefix + "WaitForPlugins")
         super().__init__(prefix, name=name)
 
 
+async def arm_and_trigger_detector_and_check_status_pv(
+    driver: ADBase,
+    good_states: Set[DetectorState] = DEFAULT_GOOD_STATES,
+) -> AsyncStatus:
+    status = await set_and_wait_for_value(driver.acquire, True)
+
+    async def completion_task() -> None:
+        await status
+        state = await driver.detector_state.get_value()
```

Comment:
This might work for now, but there is some code in Malcolm to cope with the race condition between the callback completing and the state updating:
https://github.com/dls-controls/pymalcolm/blob/d46319e9ea6b95c0856bfc2fe930b500b2f7949c/malcolm/modules/ca/parts/caactionpart.py#L98-L114
Maybe put this in as a comment that we could add this if it is a problem?